src/main/java/com/gotocompany/firehose/filter/timestamp/TimestampFilter.java

```diff
@@ -0,0 +1,322 @@
+
+        firehoseInstrumentation.captureCount(MESSAGES_PROCESSED, (long) processedCount);
+        firehoseInstrumentation.captureCount(VALID_MESSAGES, (long) validCount);
+        firehoseInstrumentation.captureCount(INVALID_MESSAGES, (long) invalidCount);
+        firehoseInstrumentation.captureValue(FILTER_DURATION_MS, (int) duration);
+
+        firehoseInstrumentation.logInfo("TimestampFilter processed {} messages in {}ms: {} valid, {} invalid, {} deserialization errors",
+                processedCount, duration, validCount, invalidCount, deserializationErrorCount);
+
+        return filteredMessages;
+    }
+
+    private boolean isValidTimestamp(DynamicMessage message) throws FilterException {
+        if (message == null) {
+            firehoseInstrumentation.logWarn("Null message provided to timestamp validation");
+            firehoseInstrumentation.captureCount(INVALID_TIMESTAMP_ERRORS, 1L);
+            return false;
+        }
+
+        try {
+            Descriptors.FieldDescriptor fieldDescriptor = message.getDescriptorForType().findFieldByName(timestampFieldName);
+
+            if (fieldDescriptor == null) {
+                firehoseInstrumentation.logWarn("Field '{}' not found in message type '{}'",
+                        timestampFieldName, message.getDescriptorForType().getFullName());
+                firehoseInstrumentation.captureCount(UNKNOWN_FIELD_ERRORS, 1L);
+                throw new FilterException("Field " + timestampFieldName + " not found in message");
+            }
+
+            if (!message.hasField(fieldDescriptor)) {
+                firehoseInstrumentation.logDebug("Message does not contain the timestamp field '{}'", timestampFieldName);
+                firehoseInstrumentation.captureCount(INVALID_TIMESTAMP_ERRORS, 1L);
+                return false;
+            }
+
+            Object fieldValue = message.getField(fieldDescriptor);
+
+            if (fieldValue == null) {
+                firehoseInstrumentation.logDebug("Timestamp field '{}' has null value", timestampFieldName);
+                firehoseInstrumentation.captureCount(INVALID_TIMESTAMP_ERRORS, 1L);
+                return false;
+            }
+
+            long timestamp = extractTimestampValue(fieldValue);
+
+            long currentTimeSeconds = Instant.now().getEpochSecond();
+            long pastThreshold = currentTimeSeconds - pastWindowSeconds;
+            long futureThreshold = currentTimeSeconds + futureWindowSeconds;
+
+            if (firehoseInstrumentation.isDebugEnabled()) {
+                firehoseInstrumentation.logDebug("Timestamp: {}, Current: {}, Past threshold: {}, Future threshold: {}",
+                        timestamp, currentTimeSeconds, pastThreshold, futureThreshold);
+            }
+
+            boolean isValid = timestamp >= pastThreshold && timestamp <= futureThreshold;
+
+            if (!isValid) {
+                firehoseInstrumentation.captureCount(INVALID_TIMESTAMP_ERRORS, 1L);
```

Comment:
We can remove this firehoseInstrumentation.

Reply:
fixed in commit - https://github.com/goto/firehose/commit/18bdfa7a24c4e6266a90b297fe32b0d73ff50aac